src/plugins/data/server/search/strategies/es_search/request_utils.ts

```diff
@@ -32,5 +33,7 @@ export async function getDefaultSearchParams(
       maxConcurrentShardRequests > 0 ? maxConcurrentShardRequests : undefined,
     ignore_unavailable: true, // Don't fail if the index/indices don't exist
     track_total_hits: true,
+    /** @removeBy 9.0 */
+    enable_fields_emulation: true, // See https://github.com/elastic/elasticsearch/pull/75745
```

Comment:
Quick remark from my side: turning the flag on doesn't do anything any more on 8.0 and 8.1 except not erroring.
The reason is that CCS support from 8.0 should only reach back to 7.last, which doesn't need to emulate the "fields" behaviour. Unless I'm mistaken about the meaning of the backward compatibility, this means you don't need the flag on 8.0 and 8.1 I think.